storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw_test.go

```diff
@@ -235,6 +237,356 @@ func TestFromMetrics(t *testing.T) {
 	})
 }
 
+func TestTemporality(t *testing.T) {
+	ts := time.Unix(100, 0)
+
+	tests := []struct {
+		name           string
+		allowDelta     bool
+		convertToNHCB  bool
+		inputSeries    []pmetric.Metric
+		expectedSeries []prompb.TimeSeries
+		expectedError  string
+	}{
+		{
+			name:       "all cumulative when delta not allowed",
+			allowDelta: false,
+			inputSeries: []pmetric.Metric{
+				createOtelSum("test_metric_1", pmetric.AggregationTemporalityCumulative, ts),
+				createOtelSum("test_metric_2", pmetric.AggregationTemporalityCumulative, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromCounter("test_metric_1", ts),
+				createPromCounter("test_metric_2", ts),
+			},
+		},
+		{
+			name:       "all delta when allowed",
+			allowDelta: true,
+			inputSeries: []pmetric.Metric{
+				createOtelSum("test_metric_1", pmetric.AggregationTemporalityDelta, ts),
+				createOtelSum("test_metric_2", pmetric.AggregationTemporalityDelta, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromCounter("test_metric_1", ts),
+				createPromCounter("test_metric_2", ts),
+			},
+		},
+		{
+			name:       "mixed temporality when delta allowed",
+			allowDelta: true,
+			inputSeries: []pmetric.Metric{
+				createOtelSum("test_metric_1", pmetric.AggregationTemporalityDelta, ts),
+				createOtelSum("test_metric_2", pmetric.AggregationTemporalityCumulative, ts),
+			},
+			expectedSeries: []prompb.TimeSeries{
+				createPromCounter("test_metric_1", ts),
+				createPromCounter("test_metric_2", ts),
+			},
+		},
+		{
+			name:          "delta rejected when not allowed",
+			allowDelta:    false,
+			expectedError: `invalid temporality and type combination for metric "test_metric_2"`,
+			inputSeries: []pmetric.Metric{
+				createOtelSum("test_metric_1", pmetric.AggregationTemporalityCumulative, ts),
+				createOtelSum("test_metric_2", pmetric.AggregationTemporalityDelta, ts),
+			},
```

Comment:
Shouldn't we expect the cumulative counter at least?

Reply:
You're right. Previously, if there was an expected error, the test would not check for the expectedSeries as well

Fixed test expectations and updated test cases in https://github.com/prometheus/prometheus/pull/16360/commits/0c25109c3772ab65876f69d1549044f5e64c2f7c